app/assets/javascripts/change_password.js

```diff
@@ -0,0 +1,16 @@
+jQuery(function($) {
+    $('#new_password_confirmation, #new_password, #password').keyup(function() {
+      var passwordPresent = $('#password').val().length > 0;
+      var button = $('input[name="commit"]');
+      var newPass = $('#new_password').val();
+      var newPassConfirm = $('#new_password_confirmation').val();
+      var newPasswordsMatches = newPass === newPassConfirm;
+      var newPasswordsPresent = (newPass.length > 0) && (newPassConfirm.length > 0);
+
+      if  (passwordPresent && newPasswordsMatches && newPasswordsPresent) {
+        button.removeAttr('disabled');
+      } else {
+        button.attr('disabled','disabled');
+      }
+    });
+});
```

Comment:
This works in preventing the user from submitting the form before having entered all the data and in ensuring that the user has entered the same value for the "new_password" and "new_password_confirmation" fields.

The problem from a user perspective is that he is never informed about what to do in order to get the field enabled. Most people will understand that they have to enter their old password and their new password two times, so this might not be the worst of problems. But as the fields as password fields, the values aren't shown. The user has therefore no way of knowing that he inserted different values for "new_password" and "new_password_confirmation". The button will simply remain disabled.

But in order to do that, we would have to start showing client side error messages. In order to do this, we would have to use angularjs here.

From a technical perspective, this seems to be a lot of code that repeats browser behaviour. One step, that would e.g. ensure to have all fields set, is to set the "required" attribute on the form input fields.